Userland/Libraries/LibGfx/ImageFormats/ExifOrientedBitmap.h

```diff
@@ -32,6 +33,17 @@ class ExifOrientedBitmap {
         return ExifOrientedBitmap(move(bitmap), size, orientation);
     }
 
+    static Gfx::TIFF::Orientation extract_orientation_from_exif(ReadonlyBytes exif_data)
+    {
+        auto exif_metadata_result = Gfx::TIFFImageDecoderPlugin::read_exif_metadata(exif_data);
+        if (exif_metadata_result.is_error())
+            return Gfx::TIFF::Orientation::Default;
+
+        auto& exif_metadata = exif_metadata_result.value();
+
+        return exif_metadata->orientation().value_or(Gfx::TIFF::Orientation::Default);
+    }
```

Comment:
Instead of adding this to the `.h` file causing it to be compiled for all individual units, please add a separate `.cpp` for `ExifOrientedBitmap` and define it out of line.

Reply:
Ok, I should also probably move rotate_bitmap there